README.md

```diff
@@ -111,6 +111,19 @@ checkerFramework {
 The check for test targets is entirely syntactic: this option will not apply the checkers
 to any task whose name includes "test", ignoring case. The default value is `false`.
 
+You can also choose different versions of the annotations for the source and
```

Comment:
Why would you want to do this?
Although it's true, I would not advertise it.  Let's keep the documentation shorter and focused on the usual use case.  I think this will be obvious to people who are using the plugin and need this capability.

Reply:
This goes back to which of the two work-arounds/fixes we expect users to use. I added this because you mentioned that you'd like both options to be available (i.e. you like fix2 better than fix1).

However, a closer look at https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph, which is Gradle's docs describing configurations, has convinced me that I did this incorrectly.

In particular, the `compile` and `testCompile` configurations are deprecated, and we shouldn't be encouraging users to use them. Both also leak onto the run-time classpath (unlike `compileOnly`). I've therefore updated both the code and the docs to suggest that users only use those configurations, and included a note that says that `testCompileOnly` is only needed when tests contain annotations.

Reply:
I like fix 2 better but I think we should only support one of them.
I didn't know about testCompileOnly and I agree it's better than testCompile.

Reply:
I believe what's here now corresponds exactly to fix2, with `testCompile` replaced by `testCompileOnly`. So this change should (hopefully) work for you. Can you confirm that, and then merge this PR if so?